doc/guides/device_mgmt/index.rst

```diff
@@ -39,14 +39,466 @@ functionality over BLE and serial.
+``STATS_SECT_ENTRY32``), ``STATS_SECT_ENTRY16`` or ``STATS_SECT_ENTRY64``.
+All statistics in a section must be declared with the same size.
+
+The statistics counters can either have names or not, depending on the setting
+of the :option:`CONFIG_STATS_NAMES` option. Using names requires an extra
+declaration step::
+
+  STATS_NAME_START(my_stats)
+    STATS_NAME(my_stats, my_stat_counter1)
+    STATS_NAME(my_stats, my_stat_counter2)
+    STATS_NAME(my_stats, my_stat_counter3)
+  STATS_NAME_END(my_stats);
+
+.. tip::
+
+   Disabling :option:`CONFIG_STATS_NAMES` will free resources. When this option
+   is disabled the ``STATS_NAME*`` macros output nothing, so adding them in the
+   code does not increase the binary size.
+
+.. tip::
+
+   :option:`CONFIG_STAT_MGMT_MAX_NAME_LEN` sets the maximum length of a section
+   name that can can be accepted as parameter for showing the section data, and
+   might require tweaking for long section names.
+
+The final steps to use a statistics section is to initialize and register it::
+
+  rc = STATS_INIT_AND_REG(my_stats, STATS_SIZE_32, "my_stats");
+  assert (rc == 0);
+
+In the running code a statistics counter can be incremented by 1 using
+``STATS_INC``, by N using ``STATS_INCN`` or reset with ``STATS_CLEAR``.
+
+Let's suppose we want to increment those counters by ``1``, ``2`` and ``3``
+every second. To get a list of stats::
+
+  $ mcumgr --conn acm0 stat list
+  stat groups:
+    my_stats
+
+To get the current value of the counters in ``my_stats``::
+
+  $ mcumgr --conn acm0 stat my_stats
+  stat group: my_stats
+        13 my_stat_counter1
+        26 my_stat_counter2
+        39 my_stat_counter3
+
+  $ mcumgr --conn acm0 stat my_stats
+  stat group: my_stats
+        16 my_stat_counter1
+        32 my_stat_counter2
+        48 my_stat_counter3
+
+When :option:`CONFIG_STATS_NAMES` is disabled the output will look like this::
+
+  $ mcumgr --conn acm0 stat my_stats
+  stat group: my_stats
+         8 s0
+        16 s1
+        24 s2
+
+.. _fs_mgmt:
+
+Filesystem Management
+*********************
+
+The filesystem module is disabled by default because of security concerns,
+because of a lack of access control every file in the FS will be accessible,
+including secrets, etc. To enable it :option:`CONFIG_MCUMGR_CMD_FS_MGMT` must
+be set (``y``). Once enabled the following sub-commands can be used::
+
+  mcumgr <connection-options> fs download <remote-file> <local-file>
+  mcumgr <connection-options> fs upload <local-file> <remote-file>
+
+Using the ``fs`` command, requires :option:`CONFIG_FILE_SYSTEM` to be enabled,
+and that some particular filesystem is enabled and properly mounted by the running
+application, eg for littefs this would mean enabling
+:option:`CONFIG_FILE_SYSTEM_LITTLEFS`, defining a storage partition :ref:`flash_map_api`
+and mounting the filesystem in the startup (:c:func:`fs_mount`).
+
+Uploading a new file to a littlefs storage, mounted under ``/lfs`` can be done with::
```

Comment:
```suggestion
Uploading a new file to a littlefs storage, mounted under ``/lfs``, can be done with::
```
I think that  `mounted under "/lfs"` is a week interjection.

Reply:
But could it be a "weekend outerjection" instead! Done.